collectors/python.d.plugin/logind/logind.chart.py

```diff
@@ -3,83 +3,165 @@
+            ['sessions_closing', 'Closing', 'absolute', 1, 1]
         ]
     },
-    'seats': {
-        'options': [None, 'Logind Seats', 'seats', 'seats', 'logind.seats', 'line'],
+    'users': {
+        'options': [None, 'Logind Users', 'users', 'users', 'logind.users', 'stacked'],
         'lines': [
-            ['seats', 'Active Seats', 'absolute', 1, 1]
+            ['users_offline', 'Offline', 'absolute', 1, 1],
+            ['users_lingering', 'Lingering', 'absolute', 1, 1],
+            ['users_online', 'Online', 'absolute', 1, 1],
+            ['users_active', 'Active', 'absolute', 1, 1],
+            ['users_closing', 'Closing', 'absolute', 1, 1]
         ]
-    }
+    },
 }
 
 
-class Service(ExecutableService):
+class Service(SimpleService):
     def __init__(self, configuration=None, name=None):
-        ExecutableService.__init__(self, configuration=configuration, name=name)
+        SimpleService.__init__(self, configuration=configuration, name=name)
         self.order = ORDER
         self.definitions = CHARTS
-        self.command = LOGINCTL_COMMAND
+        self.bus_name = configuration.get('bus_name', BUS_NAME)
+        self.obj_path = configuration.get('object_path', OBJ_PATH)
+        self.bus = dbus.SystemBus()
+        self.mgr = None
+        self.iface = None
+
+        try:
+            self.mgr = self.bus.get_object(self.bus_name, self.obj_path)
+            self.iface = dbus.Interface(self.mgr, MGR_IFACE)
+        except dbus.DBusException:
+            pass
+
+    def check(self):
+        if not HAVE_DBUS:
+            self.error('dbus-python is not available')
+            return False
+
+        if self.mgr is None:
+            self.error('Unable to connect to ' + self.bus_name + ' service')
+            return False
+
+        if self.iface is None:
+            self.error('Failed to construct interface')
+
+        return self._get_data()
 
     def _get_data(self):
+        try:
+            sessions = self.iface.ListSessions()
+            users = self.iface.ListUsers()
+        except dbus.DBusException:
+            return None
+
         ret = {
             'sessions_graphical': 0,
             'sessions_console': 0,
             'sessions_remote': 0,
+            'sessions_other': 0,
+            'sessions_online': 0,
+            'sessions_active': 0,
+            'sessions_closing': 0,
+            'users_offline': 0,
+            'users_lingering': 0,
+            'users_online': 0,
+            'users_active': 0,
+            'users_closing': 0,
         }
-        users = {
-            'graphical': list(),
-            'console': list(),
-            'remote': list()
-        }
-        seats = list()
-        data = self._get_raw_data()
 
-        for item in data:
-            fields = item.split()
-            if len(fields) == 3:
-                users['remote'].append(fields[2])
+        session_paths = {s[4] for s in sessions}
+        user_paths = {u[2] for u in users}
```

Comment:
What are those 4 and 2? Perhaps add a comment with example output

Reply:
Agreed.